### ISSUE TITLE
mypy_path option split on [:,] before environment variable expansion

### ISSUE THREAD

**Issue by PymZoR** [python/mypy#9214]
Hi,

given an environment variable composed of multiple path, eg `export A=/foo:/bar`, using the following mypy.ini fails to recognize the two paths `foo` and `bar`
```
[mypy]
mypy_path = $A
```

This is a consequence of splitting the option on [:,] before expanding environment variables. (https://github.com/python/mypy/blob/master/mypy/config_parser.py#L85)

Would it be sufficient / not harmful to replace the parsing funciton with the following ?
```python
lambda s: re.split('[,:]', expand_path(s.strip()))
```

**Comment by JelleZijlstra** [comment]
I agree that we should expand env vars before splitting. Your proposed fix would not quite work though, because we also do `os.path.expanduser`, and that doesn't work properly on an unsplit path:
```
In [5]: os.path.expanduser('/my/stubs:~/stubs')                                                                                                                                      
Out[5]: '/my/stubs:~/stubs'
```
So the solution might require a bit more refactoring around the `expand_path()` function.

**Comment by wyattscarpenter** [comment]
Could be as easy as

```py
"mypy_path": lambda s: [os.path.expanduser(p.strip()) for p in re.split("[,:]", os.path.expandvars(s))],
```

then, if I understand right. Plus documentation tweak and tests.
